### PR TITLE
MCKIN-23062 purge cohorts of deleted groups

### PR DESCRIPTION
**Pull request by mudassir-hafeez**
This management command will purge cohorts of deleted groups for specific courses

### CODE REVIEW

File: edx_solutions_projects/management/commands/purge_deleted_groups_cohorts.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+            if op.lower() != 'remove':
+                raise Exception("Only 'remove' operations supported")
+
+            course_keys = [get_course_key(course_id) for course_id in courses]
+            if None in course_keys:
+                raise Exception("At least one course failed to create course key")
+
+            # Default cohort must exclude e.g default_cohort
+            exclude_cohorts = [CourseUserGroup.default_cohort_name]
+            projects = Project.objects.filter(course_id__in=courses).prefetch_related('workgroups')
```
Comment (nasirhjafri) [inline edx_solutions_projects/management/commands/purge_deleted_groups_cohorts.py]: Why don't we query on workgroups? `Workgroup.objects.filter(project__course_id__in=courses)`

File: edx_solutions_projects/management/commands/purge_deleted_groups_cohorts.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+            for project in projects:
+                for work_group in project.workgroups.all():
+                    if work_group.project.course_id in courses:
```
Comment (nasirhjafri) [inline edx_solutions_projects/management/commands/purge_deleted_groups_cohorts.py]: Isn't the same condition in query and projects are already filtered?

File: edx_solutions_projects/management/commands/purge_deleted_groups_cohorts.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+                        exclude_cohorts.append(work_group.cohort_name)
+
+            total_obsoleted_cohorts = 0
+            obsoleted_cohorts = CourseUserGroup.objects.filter(
+                course_id__in=course_keys,
+                group_type=CourseUserGroup.COHORT
+            ).exclude(name__in=exclude_cohorts)
+            total_obsoleted_cohorts = obsoleted_cohorts.count()
+            obsoleted_cohorts.delete()
```
Comment (nasirhjafri) [inline edx_solutions_projects/management/commands/purge_deleted_groups_cohorts.py:36]: `.delete()` should return the number of deleted rows.

**Review by nasirhjafri — CHANGES_REQUESTED** [review]

File: edx_solutions_projects/management/commands/purge_deleted_groups_cohorts.py (hunk 1 above)
Comment (mudassir-hafeez) [reply edx_solutions_projects/management/commands/purge_deleted_groups_cohorts.py]: Addressed

**Review by nasirhjafri — APPROVED** [review]

**Pull request merged**